lib/dslash_pack2.cu

```diff
@@ -63,55 +64,53 @@ protected:
     const double b;
     const double c;
     int twist; // only has meaning for nSpin=4
+    int shmem;
 
     bool tuneGridDim() const { return true; } // If striping, always tune grid dimension
 
     unsigned int maxGridSize() const
     {
-      if (location & Host) {
+      if (location & Host || location & Shmem) {
 #ifdef STRIPED
         // if zero-copy policy then set a maximum number of blocks to be
         // the 3 * number of dimensions we are communicating
         int max = 3;
 #else
         // if zero-copy policy then assign exactly up to four thread blocks
         // per direction per dimension (effectively no grid-size tuning)
-        int max = 2 * 4;
+        int max = ((shmem & 32) || (location & Host)) ? 4 : -1;
```

Comment:
What's the reason for the `|| (location & Host)`? Was some issue with the previous default of `8` when `STRIPED` was undefined uncovered with `Host` specifically?

Reply:
Alternatively, is in related to the changes in `minGridSize` below that I just noticed?

Reply:
Above if has changed (now line 73). This no longer uniquely is always host and we only limit the maxGridsize for zero copy and nvshmem uber kernel (shmem & 32).